sklego/meta/ordinal_classification.py

```diff
@@ -0,0 +1,233 @@
+        Whether or not to calibrate the binary classifiers using `CalibratedClassifierCV`.
+    calibrarion_kwargs : dict | None, default=None
+        Keyword arguments to the `CalibratedClassifierCV` class, used only if `use_calibration=True`.
+
+    Attributes
+    ----------
+    estimators_ : dict[int, scikit-learn compatible classifier]
+        The fitted underlying binary classifiers.
+    classes_ : np.ndarray of shape (n_classes,)
+        The classes seen during `fit`.
+    n_features_in_ : int
+        The number of features seen during `fit`.
+
+    Examples
+    --------
+    ```py
+    import pandas as pd
+
+    from sklearn.linear_model import LogisticRegression
+    from sklearn.model_selection import train_test_split
+
+    from sklego.meta import OrdinalClassifier
+
+    url = "https://stats.idre.ucla.edu/stat/data/ologit.dta"
+    df = pd.read_stata(url).assign(apply_codes = lambda t: t["apply"].cat.codes)
+
+    target = "apply_codes"
+    features = [c for c in df.columns if c not in {target, "apply"}]
+
+    X, y = df[features].to_numpy(), df[target].to_numpy()
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+
+    clf = OrdinalClassifier(LogisticRegression(), n_jobs=-1)
+    _ = clf.fit(X_train, y_train)
+    clf.predict_proba(X_test)
+    ```
+
+    Notes
+    -----
+    The implementation is based on the paper [A simple approach to ordinal classification](https://www.cs.waikato.ac.nz/~eibe/pubs/ordinal_tech_report.pdf)
+    by Eibe Frank and Mark Hall.
+
+    """
+
+    def __init__(self, estimator, *, n_jobs=None, use_calibration=False, **calibrarion_kwargs):
+        self.estimator = estimator
+        self.n_jobs = n_jobs
+        self.use_calibration = use_calibration
+        self.calibrarion_kwargs = calibrarion_kwargs
+
+    def fit(self, X, y):
+        """Fit the `OrdinalClassifier` model on training data `X` and `y` by fitting its underlying estimators on
+        N-1 datasets `X` and `y` for each class `y_label` in `y` except `y.max()`.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features )
+            The training data.
+        y : array-like of shape (n_samples,)
+            The target values.
+
+        Returns
+        -------
+        self : OrdinalClassifier
+            Fitted model.
+
+        Raises
+        ------
+        ValueError
+            If the estimator is not a classifier or if it does not implement `.predict_proba()`.
+        """
+
+        if not is_classifier(self.estimator):
+            raise ValueError("The estimator must be a classifier.")
+
+        if not hasattr(self.estimator, "predict_proba"):
+            raise ValueError("The estimator must implement `.predict_proba()` method.")
+
+        X, y = check_X_y(X, y, estimator=self)
+
+        self.classes_ = np.sort(np.unique(y))
+        self.n_features_in_ = X.shape[1]
+
+        if self.n_classes_ < 2:
```

Comment:
I wonder ... don't we need at least three classes for ordinal regression to make sense? Any binary classification with two classes is automatically ordinal, no?

Reply:
Also very true! I cheated for the scikit-learn tests 😂 
Let me figure out how to deal with those